mpu6050/mpu6050.c

```diff
@@ -55,33 +104,91 @@ static int mpu6050_read_data(void)
 		return -EINVAL;
 	}
 
-	for (i = 0, p = g_mpu6050_data.data.raw; i < READ_REG_NUM * sizeof(u16); i += 2, ++p)
+	head = get_next_pos(g_mpu6050_data.head, READ_DEPTH);
+
+	for (i = 0, p = g_mpu6050_data.data[head].raw; i < READ_REG_NUM * sizeof(u16); i += 2, ++p)
 		*p = (s16) ((u16) values[i] << 8 | (u16)values[i + 1]);
 
+	spin_lock(&g_mpu6050_data.lock);
+	g_mpu6050_data.head = head;
+	count = g_mpu6050_data.count;
+	g_mpu6050_data.count = count + 1;
+	spin_unlock(&g_mpu6050_data.lock);
+
+	dev_info(&drv_client->dev, "sensor data read head %d, count %d\n", g_mpu6050_data.head, g_mpu6050_data.count);
+
 	/* Temperature in degrees C =
 	 * (TEMP_OUT Register Value  as a signed quantity)/340 + 36.53
 	 */
-	temp = g_mpu6050_data.data.values.temperature;
-	g_mpu6050_data.data.values.temperature = (temp + 12420 + 170) / 340;
+	temp = g_mpu6050_data.data[head].values.temperature;
+	g_mpu6050_data.data[head].values.temperature = (temp + 12420 + 170) / 340;
+
+	if ((is_buf && count < READ_DEPTH) || (!is_buf && !count))
+		complete(&has_data);
```

Comment:
Just to note: I haven't made deep analyse bu afraid of possibility of some race with `mpu6050_get_data()`,
which may cause `has_data` became desynchronized with `g_mpu6050_data.count`.